src/Shapes.hs

```diff
@@ -197,6 +197,36 @@ instance Named Square where
 instance ToJSON Square
 instance FromJSON Square
 
+-------------------------- (Angle Mark, added by Dor)
+data AngleMark = AngleMark { xam :: Float -- Starting point for angle mark
+                     , yam :: Float
+                     , radiusam :: Float -- The radius of the angle mark
+                     , sizeam :: Float
+                     , nameam :: String
+                     , coloram :: Color
+                     , angleam :: Float -- The angle that the angle mark specifies
+                     , rotationam :: Float -- The rotation angle of amgle mark
+                     , isRightam :: String -- Is this a right angle or not 
```

Comment:
This is probably better as a bool or an enum `data AngleType = Right | NotRtAngle`

Reply:
Yes I know, I had trouble doing that at the begging but now I think I know how to solve that